core/dbt/contracts/graph/compiled.py

```diff
@@ -119,12 +119,7 @@ class CompiledSchemaTestNode(CompiledNode, HasTestMetadata):
     column_name: Optional[str] = None
     config: TestConfig = field(default_factory=TestConfig)
 
-    def same_config(self, other) -> bool:
-        return (
-            self.unrendered_config.get('severity') ==
-            other.unrendered_config.get('severity')
-        )
-
+    # TODO: this is unused
```

Comment:
Yes... indeed it is. I confirmed that `dbt test -m state:modified` works as expected, leveraging the `unrendered_config`.